ui/tests/e2e/integration/settings.spec.js

```diff
@@ -0,0 +1,71 @@
+import { expect, test } from '@playwright/test';
+import { ENV } from '../env';
+
+const verifyAdapterResBody = (body) => {
+  expect(body).toBeInstanceOf(Array);
+  body.forEach(({ adapter_location, name, version, git_commit_sha, ops }) => {
+    expect(adapter_location).toMatch(/localhost:\d+/);
+    expect(name).toEqual(expect.any(String));
+    expect(version).toEqual(expect.any(String));
+    expect(git_commit_sha).toEqual(expect.any(String));
+    // ops can be null or array
+    if (ops) {
+      expect(ops).toBeInstanceOf(Array);
+      ops.forEach(({ key, value, category }) => {
+        expect(key).toEqual(expect.any(String));
+        expect(value).toEqual(expect.any(String));
+        // category may or may not be present, if present its an integer
+        if (category) {
+          expect(category).toEqual(expect.any(Number));
+        } else {
+          expect(category).toBeUndefined();
+        }
+      });
+    } else {
+      expect(ops).toBeNull();
+    }
+  });
+};
+
+test.describe('Settings', () => {
+  test.describe('Service Meshes', () => {
```

Comment:
we shouldnt have nested describe blocks .

Reply:
The nested describe blocks were used for better grouping of tests. Like, using a single describe for a page we can group all tests of that page under it. But say the page has different sections, and we write multiple tests for each section. The for a `Page > Section > Test` like grouping, nested describe blocks have been used.

In this case currently there is only one test, but in the future more might get added. The nested describe blocks may help in grouping and organizing them at that time.

Reply:
Done. Refactored the tests to avoid usage of nested describe blocks.